docs/source/rag/retrieval_question_generation.rst

```diff
@@ -0,0 +1,208 @@
+.. _retrieval-question-generation:
+
+====================================
+Question Generation for Retrieval Evaluation (Experimental)
+====================================
+
+MLFlow provides a state-of-the-art experience for build Retrieval-Augmented Generation (RAG) models. 
+RAG is a cutting edge approach that combines the strengths of retrieval models and generative models.
+It effectively merges the capabilities of searching and generating text to provide more contextually
+relevant and coherent responses to questions. RAG leverges a retriever to find context documents, and
+this novel approach has revolutionized various NLP tasks.
+
+Naturally, we want to be able to evaluate this retriever system for the RAG model to compare and judge its
+performance. To evaluate a retriever system, we would first need a test set of questions on the documents.
+These questions need to be diverse, relevant, and coherent. Manually generating questions may be challenging
+because it first requires you to understand the documents, and spend lots of time coming up with questions 
+for them. 
+
+We want to make this process simpler by utilizing an LLM to generate questions for this test set. This
+tutorial will walk through how to generate the questions and how to analyze the diversity and relevance
+of the questions.
+
+.. _retrieval-question-generation-quickstart:
+
+The following guide will walk you through generating the question test set for retriever evaluation.
+
+Step 1: Install and Load Packages
+==================================
+
+.. code-block:: python
+   import os
+   import pickle
+   from langchain.docstore.document import Document
+   from langchain.embeddings import OpenAIEmbeddings
+   import openai
+
+Step 2: Load Data
+==================
+Load the data through Langchain Documents to utilize their embedding models.
+
+.. code-block:: python
+   from langchain.docstore.document import Document
+   import pickle
+
+   docs, metadatas = None, None
+   with open("/dbfs/bbqiu/mlflow_docs.pkl", "rb") as f:
+   docs = pickle.load(f)
+   with open("/dbfs/bbqiu/mlflow_docs_metadata.pkl", "rb") as f:
```

Comment:
Definitely shouldn't hardcode @bbqiu 's DBFS paths here. This data isn't available to customers.

Reply:
Yep, I'll make the change for this and polish up the other stuff too. For the current rough draft I just wanted to make sure the content was what we want before polishing, since its common for tutorial contents to change